### PR TITLE
Force the recompute of the name during the copy

### PR DESCRIPTION
**Pull request by jssuzanne**
See the bug report https://github.com/OCA/partner-contact/issues/72

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Maybe we can just add copy=False in field definition?

**Comment by jssuzanne** [comment]
What append if a field with copy=False is put in the default value of the copy_data method ?
I haven't been the test.

**Comment by jssuzanne** [comment]
:O sorry for the flake8

**Comment by jssuzanne** [comment]
I have fix the flake8 in my branch how update the pull request

File: partner_firstname/partner.py [hunk 1]
```diff
@@ -104,8 +104,10 @@ def copy_data(self, cr, uid, _id, default=None, context=None):
                     cr, uid, [_id], ['lastname'], context=context
                     )[0]['lastname']
             )
-            if default.get('name'):
-                del(default['name'])
+        # force the recompute of the name ater the copy
```
Comment (eLBati) [inline partner_firstname/partner.py]: ater/after

File: partner_firstname/res_user.py [hunk 2]
```diff
@@ -37,7 +38,9 @@ def copy_data(self, cr, uid, _id, default=None, context=None):
                     cr, uid, [_id], ['lastname'], context=context
                 )[0]['lastname']
             )
-            if default.get('name'):
-                del(default['name'])
+        # force the recompute of the name ater the copy
```
Comment (eLBati) [inline partner_firstname/res_user.py]: ater/after

**Comment by yvaucher** [comment]
@jssuzanne a field defined with copy=False won't be copied when duplicating a record.

File: partner_firstname/res_user.py [hunk 3]
```diff
@@ -25,6 +25,7 @@ class ResUsers(orm.Model):
 
     _inherit = 'res.users'
 
+    # We should use an AbstractModel to not write two time the same method
```
Comment (joaoalf) [inline partner_firstname/res_user.py:28]: Hello @jssuzanne 

I'm not sure I understood your code comment. Can you clarify?

**Comment by StefanRijnhart** [comment]
Bug fixed upstream in https://github.com/odoo/odoo/commit/3a44d84b0f0dd7a0596f4854652391538e7a764e

**Pull request closed without merging**